tests/test_validation_framework/test_upscaling.py

```diff
@@ -0,0 +1,120 @@
+# Copyright (c) 2021,Vienna University of Technology,
+# Department of Geodesy and Geoinformation
+# All rights reserved.
+
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#   * Redistributions of source code must retain the above copyright notice,
+#     this list of conditions and the following disclaimer.
+#   * Redistributions in binary form must reproduce the above copyright notice,
+#     this list of conditions and the following disclaimer in the documentation
+#     and/or other materials provided with the distribution.
+#   * Neither the name of the Vienna University of Technology, Department of
+#     Geodesy and Geoinformation nor the names of its contributors may be used
+#     to endorse or promote products derived from this software without specific
+#     prior written permission.
+
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL VIENNA UNIVERSITY OF TECHNOLOGY, DEPARTMENT OF
+# GEODESY AND GEOINFORMATION BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
+# BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+# IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+"""Test for the data averaging class"""
+
+import warnings
+
+import pandas as pd
+import numpy as np
+from datetime import datetime
+
+from pytesmo.validation_framework.upscaling import Upscaling
+
+with warnings.catch_warnings():
+    warnings.filterwarnings("ignore")
+
+
+def return_averager():
+    manager_parms = {
+        "datasets": None,
+        "period": None,
+        "read_ts_names": None,
+    }
+    averager = Upscaling(
+        ref_class=None,
+        others_class={},
+        upscaling_lut={},
+        manager_parms=manager_parms
+    )
+
+    return averager
```

Comment:
You could make this a pytest.fixture:
```
@pytest.fixture
def averager():
    ...
    return averager
```

and then use it as function argument for the tests:

```
def test_upscale(averager):
    ...
    upscaled = averager.upscale(to_upscale, method="average")
    ...
```